lib/promauth/config.go

```diff
@@ -823,3 +875,113 @@ func (tctx *tlsContext) initFromTLSConfig(baseDir string, tc *TLSConfig) error {
 	tctx.minVersion = v
 	return nil
 }
+
+// NewTLSRoundTripper returns new http.RoundTripper which automatically updates
+// RootCA in tls.Config whenever it changes.
+func (tctx *tlsContext) NewTLSRoundTripper(cfg *tls.Config, builder func(transport *http.Transport)) (http.RoundTripper, error) {
+	// TLS context is not initialized so use the provided RoundTripper without wrapper
+	if tctx == nil || tctx.getRootCAPEM == nil {
+		tr := &http.Transport{
+			TLSClientConfig: cfg,
+		}
+		builder(tr)
+		return tr, nil
+	}
+
+	var mu sync.Mutex
+	var deadline uint64
+	var rootCA []byte
+	var err error
+	getTLSDigests := func() []byte {
+		if fasttime.UnixTimestamp() > deadline {
+			mu.Lock()
```

Comment:
I think mutex must be acquired before if statement.

Reply:
Fixed this